### PR TITLE
[WFCORE-3187] add-user.sh tool prints unexpected characters if arrow keys are pushed during interactive mode

### PR DESCRIPTION
**Pull request by MMarus**
JBEAP: https://issues.jboss.org/browse/JBEAP-6640
WFCORE: https://issues.jboss.org/browse/WFCORE-3187

Please DON'T merge before https://github.com/wildfly/wildfly-core/pull/2533 is merged, since this pull request requires changes introduced in https://github.com/wildfly/wildfly-core/pull/2533 .

### CODE REVIEW

**Comment by MMarus** [comment]
The first 9 commits are from https://github.com/wildfly/wildfly-core/pull/2533 . The https://github.com/wildfly/wildfly-core/pull/2715/commits/526485a4113b958b877a91d0e9839ecc2c4a6d73 addresses the WFCORE-3187 and JBEAP-6640 issues which are subjects of this PR.

File: domain-management/src/main/java/org/jboss/as/domain/management/security/adduser/JavaConsole.java [hunk 1]
```diff
@@ -59,26 +61,34 @@ public void printf(String format, Object... args) throws IllegalFormatException
...
-            throw ROOT_LOGGER.noConsoleAvailable();
+        ReadLineHandler readLineHandler = new ReadLineHandler(new Prompt(String.format(fmt, args)));
+        return readInputLine(readLineHandler);
+    }
+
+    private String readInputLine(ReadLineHandler readLineHandler) {
+        try {
+            createTerminalConnection(readLineHandler);
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (xstefank) [inline domain-management/src/main/java/org/jboss/as/domain/management/security/adduser/JavaConsole.java]: this should log a proper error - you can find an example on the line 65 before change

File: domain-management/src/test/java/org/jboss/as/domain/management/security/adduser/AeshJavaConsoleTest.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        aeshJavaConsoleTestResource.write("Foo".getBytes());
+        aeshJavaConsoleTestResource.write(Key.CTRL_C.getKeyValuesAsString().getBytes());
+        aeshJavaConsoleTestResource.flushWrites();
+
+        Thread.sleep(100);
+        assertEquals(false, aeshJavaConsoleTestResource.isAlive());
+        assertEquals(Thread.State.TERMINATED, aeshJavaConsoleTestResource.getState());
+    }
+
+    public class AeshJavaConsoleTestResource extends Thread {
```
Comment (xstefank) [inline domain-management/src/test/java/org/jboss/as/domain/management/security/adduser/AeshJavaConsoleTest.java]: public inner classes are generally considered as not a good practice, is there any reason this class needs to be public? Also please make it static

**Comment by darranl** [comment]
I think this one will be easier to review after a rebase.

**Comment by MMarus** [comment]
I moved this PR to a new rebased PR https://github.com/wildfly/wildfly-core/pull/3064 for easier review.

**Pull request closed without merging**